cloud/scope/powervs_machine_test.go

```diff
@@ -134,6 +139,116 @@ func newDHCPServerDetails(serverID, leaseIP, instanceMac string) *models.DHCPSer
 	}
 }
 
+func TestAPIServerPort(t *testing.T) {
```

Comment:
@GunaKKIBM, for simple functions where we are checking only for a certain field, any specific reason for setting up the scope via `setupPowerVSMachineScope` and creating mock client which might not be used? wondering if we can simplify into something like this, wdyt?

```
func TestAPIServerPort(t *testing.T) {
	testCases := []struct {
		name               string
		expectedportNumber int32
		machineScope       PowerVSMachineScope
	}{
		{
			name:               "test assigned port number",
			expectedportNumber: int32(6445),
			machineScope: PowerVSMachineScope{
				Cluster: &capiv1beta1.Cluster{
					Spec: capiv1beta1.ClusterSpec{
						ClusterNetwork: &capiv1beta1.ClusterNetwork{
							APIServerPort: ptr.To(int32(6445)),
						},
					},
				},
			},
		},
	}

	for _, tc := range testCases {
		g := NewWithT(t)
		t.Run(tc.name, func(_ *testing.T) {
			port := tc.machineScope.APIServerPort()
			g.Expect(port).To(Equal(tc.expectedportNumber))
		})
	}

}
```

Reply:
Agreed. Thank you! I will update this and other tests as well.

Reply:
Updated